californium-core/src/main/java/org/eclipse/californium/core/network/deduplication/CropRotation.java

```diff
@@ -48,12 +50,12 @@
 	private boolean running = false;
 	private ScheduledExecutorService executor;
 
-	private ExchangeMap[] maps;
-	private int first;
-	private int second;
+	private final ExchangeMap maps[];
+	private volatile int first;
+	private volatile int second;
```

Comment:
this seems to have been changed to allow for multi-threaded executors being used, right?
However, I think we should rather prevent multiple rotation tasks being scheduled to run in parallel instead.